### DISCUSSION TITLE
Weekly review on the page of the week.

### DISCUSSION THREAD

**Discussion by inzagi1** [YU000jp/logseq-plugin-show-weekday-and-week-number#142] [General]
Hi, thanks for creating a very useful plugin.
But I'm having some difficulties implementing the weekly review as I'm not a programmer.

I make a daily entry like this in the journal:
log:: Today was a good day #dayreview.
And these entries I want to collect on the weekly page for that week as a table (like in the image)
![ScreenShot_20240609173005](https://github.com/YU000jp/logseq-plugin-show-weekday-and-week-number/assets/172391388/1720adaa-d35b-4b29-94de-9e621adce4e9)
Can you help me with the query?

**Comment by YU000jp** [comment]
It's a good idea to summarise the day in one line.

~~The query I've found so far is this:~~ ⚠️
![image](https://github.com/YU000jp/logseq-plugin-show-weekday-and-week-number/assets/111847207/c3916497-8622-4e67-8cd4-c1bc57e31985)

```Clojure
#+BEGIN_QUERY
{
 :query [:find (pull ?b [{:block/page
     [:block/name :block/journal-day]} :block/content :block/properties])
      :where
      [?b :block/properties ?bprops]
      [(get ?bprops :log "nil") ?bs]
      [(not= ?bs "nil")]]
:result-transform (fn [result]
                     (sort-by (fn [s]
                        (get-in s [:block/page :block/journal-day])) (fn [a b] (compare b a)) result)) 
:view (fn [rows] [:table 
 [:thead 
  [:tr 
   [:th "Date"] 
   [:th "Details"] ] ] 
 [:tbody 
(for [r rows] [:tr 
   [:td (get-in r [:block/page :block/name])] 
   [:td (str/replace (get-in r [:block/content]) "log:: " "")] ])
   ]]
)
}
#+END_QUERY
```
It refers to the log property in each journal. It will be displayed even without the #dayreview tag. To change the property name, change all occurrences of the "log" string in the query. Note, this query does not support click-to-jump to blocks.

**Reply by YU000jp** [reply]
Sorry. If we just put this query in place, it would find all blocks with the log property, resulting in a long table. Need to restrict the date in the query. If possible, we need to find a way to make it appear automatically every week. It might be best to have a plugin automatically generate the week date range for the query. I will investigate whether it is possible to implement such functionality in this plugin.

**Comment by YU000jp** [comment]
I plan to systematize it and build it into this plugin.

Sample:
![image](https://github.com/YU000jp/logseq-plugin-show-weekday-and-week-number/assets/111847207/f526d951-b6f3-4bee-970f-5dcc5368c71b)
Source:
```Clojure
#+BEGIN_QUERY
{:title "Headline of each days"
 :query [:find (pull ?b [{:block/page
     [:block/name :block/journal-day]} :block/properties])
      :where
      [?b :block/properties ?bprops]
      [(get ?bprops :log "nil") ?bs]
      [(not= ?bs "nil")]
      [?b :block/page ?p]
      [?p :block/journal-day ?jd]
     [(>= ?jd 20240609)]
     [(<= ?jd 20240615)]
]
:result-transform (fn [result]
                     (sort-by (fn [s]
                        (get-in s [:block/page :block/journal-day])) (fn [a b] (compare b a)) result)) 
 :table-view? true

}
#+END_QUERY
```

**Reply by YU000jp** [reply]
I released a new version. Please try it. It is off by default, so please turn it on in the plugin settings.
